src/ripple/core/impl/ThreadEntry.cpp

```diff
@@ -22,47 +22,49 @@
 #include <ripple/basics/Log.h>
 
 #include <boost/coroutine/exceptions.hpp>
+#include <boost/thread/tss.hpp>
 #include <exception>
 #include <iostream>
 
 namespace ripple {
 
 #ifndef NO_LOG_UNHANDLED_EXCEPTIONS
 namespace detail {
-thread_local
-std::string threadName;
+boost::thread_specific_ptr<std::string> threadName;
 
 void setThreadName(std::string name)
 {
-    threadName = std::move(name);
+    threadName.reset(new std::string(std::move(name)));
 }
 }
 
 void terminateHandler()
 {
     if (std::current_exception())
     {
+        if (!detail::threadName.get())
+            detail::threadName.reset(new std::string("Unknown"));
```

Comment:
@willwray Done. I also made `threadName` static while I was at it (`threadName` used to be `extern thread_local std::string`, but there's a gcc bug that we use `setThreadName` to work around; so it should be static now).